### PR TITLE
Refs #6980 Added new examples.

### PR DESCRIPTION
**Pull request by LuisGP**
- Module example
- Iterators example

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/22/)

File: examples/iterators.cpp [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        DynamicData str1(string);
+
+        str1.value<std::string>("Hello!");
+        for (ReadableDynamicDataRef&& elem : str1)
+        {
+            std::cout << elem.value<char>();
+        }
+        std::cout << std::endl;
+
+        // Hello! -> Ifmmp"
```
Comment (lemunozm) [inline examples/iterators.cpp:22]: Ifmmp?

File: examples/iterators.cpp [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+        for (int i = 0; i < 10; ++i)
+        {
+            array[i] = 5 * i;
+        }
+
+        int32_t check_sum = 0;
+        for (ReadableDynamicDataRef&& elem : array)
+        {
+            check_sum += elem.value<int32_t>();
```
Comment (lemunozm) [inline examples/iterators.cpp:48]: Do you sure you need the `.value<int32>()`? (see ocurrences)

File: examples/iterators.cpp [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        }
+
+        for (WritableDynamicDataRef&& elem : array)
+        {
+            elem = elem.value<int32_t>() * 2;
```
Comment (lemunozm) [inline examples/iterators.cpp:53]: Same as before

File: examples/module.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    std::cout << std::boolalpha;
+    std::cout << "Does a::a::OuterType exists?: " << root.has_structure("a::a::OuterType") << std::endl;
+    std::cout << "Does ::InnerType exists?: " << root.has_structure("::InnerType") << std::endl;
+    std::cout << "Does InnerType exists?: " << root.has_structure("InnerType") << std::endl;
+    std::cout << "Does OuterType exists?: " << root.has_structure("OuterType") << std::endl;
+
+    DynamicData module_data(root["a"]["a"].structure("OuterType")); // ::a::a::OuterType
+    module_data["om1"] = "This is a string.";
+
```
Comment (lemunozm) [inline examples/module.cpp:55]: Some way to get the absolute path of a structure?

**Review by lemunozm — CHANGES_REQUESTED** [review]

File: examples/iterators.cpp (hunk 1 above)
Comment (LuisGP) [reply examples/iterators.cpp:22]: 'H' + 1 = 'I'
'e' + 1 = 'f'
etc

File: examples/iterators.cpp (hunk 2 above)
Comment (LuisGP) [reply examples/iterators.cpp:48]: Yes, because of the `operator+=` ambiguity.

File: examples/iterators.cpp (hunk 3 above)
Comment (LuisGP) [reply examples/iterators.cpp:53]: Same as before

File: examples/module.cpp (hunk 4 above)
Comment (LuisGP) [reply examples/module.cpp:55]: Already shown in the upper lines like:

    std::cout << "Does a::a::OuterType exists?: " << root.has_structure("a::a::OuterType") << std::endl;

File: examples/iterators.cpp (hunk 3 above)
Comment (lemunozm) [reply examples/iterators.cpp:53]: This has not the `operator +=`

File: examples/module.cpp (hunk 4 above)
Comment (lemunozm) [reply examples/module.cpp:55]: But this only ask about the existance, can I ask to a module to get the path?

File: examples/iterators.cpp (hunk 2 above)
Comment (lemunozm) [reply examples/iterators.cpp:48]: Can we fix it with the addition of these kinds of operators?

File: examples/iterators.cpp (hunk 3 above)
Comment (LuisGP) [reply examples/iterators.cpp:53]: Nope, but `operator*`.

File: examples/iterators.cpp (hunk 2 above)
Comment (lemunozm) [reply examples/iterators.cpp:48]: Spoken in person

File: examples/iterators.cpp (hunk 3 above)
Comment (lemunozm) [reply examples/iterators.cpp:53]: as we had talked

File: examples/iterators.cpp (hunk 1 above)
Comment (lemunozm) [reply examples/iterators.cpp:22]: :astonished:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/25/)

**Review by lemunozm — APPROVED** [review]

**Pull request merged**